src/main/java/net/sf/mzmine/modules/peaklistmethods/gapfilling/samerange/SameRangeTask.java

```diff
@@ -55,179 +59,246 @@
+                // Get current peak
+                Feature currentPeak = sourceRow.getPeak(column);
 
-	}
+                // If there is a gap, try to fill it
+                if (currentPeak == null)
+                    currentPeak = fillGap(sourceRow, column);
 
-	// Append processed peak list to the project
-	project.addPeakList(processedPeakList);
+                // If a peak was found or created, add it
+                if (currentPeak != null)
+                    newRow.addPeak(column, currentPeak);
+            }
 
-        // Add quality parameters to peaks
-	QualityParameters.calculateQualityParameters(processedPeakList);
+            outputList.add(newRow);
 
-	// Add task description to peakList
-	processedPeakList
-		.addDescriptionOfAppliedTask(new SimplePeakListAppliedMethod(
-			"Gap filling using RT and m/z range", parameters));
+            processedRowsAtomic.getAndAdd(1);
+            //processedRows++;
+        });
 
-	// Remove the original peaklist if requested
-	if (removeOriginal)
-	    project.removePeakList(peakList);
+        outputList.stream().forEach(newRow -> {
+            processedPeakList.addRow((PeakListRow)newRow);
+        });
 
-	setStatus(TaskStatus.FINISHED);
+        /* End Parallel Implementation */
+        /*******************************************************************************/
 
-	logger.info("Finished gap-filling " + peakList);
 
-    }
+        // Fill gaps in given column
+        /*for (int row = 0; row < totalRows; row++) {
+            // Canceled?
+            if (isCanceled())
+                return;
+
+            PeakListRow sourceRow = peakList.getRow(row);
+            PeakListRow newRow = new SimplePeakListRow(sourceRow.getID());
+
+            // Copy comment
+            newRow.setComment(sourceRow.getComment());
+
+            // Copy identities
+            for (PeakIdentity ident : sourceRow.getPeakIdentities())
+            newRow.addPeakIdentity(ident, false);
+            if (sourceRow.getPreferredPeakIdentity() != null)
+            newRow.setPreferredPeakIdentity(sourceRow
+                .getPreferredPeakIdentity());
+
+            // Copy each peaks and fill gaps
+            for (RawDataFile column : columns) {
+
+                // Canceled?
+                if (isCanceled())
+                    return;
+
+                // Get current peak
+                Feature currentPeak = sourceRow.getPeak(column);
+
+                // If there is a gap, try to fill it
+                if (currentPeak == null)
+                    currentPeak = fillGap(sourceRow, column);
+
+                // If a peak was found or created, add it
+                if (currentPeak != null)
+                    newRow.addPeak(column, currentPeak);
+
+            }
+
+            processedPeakList.addRow(newRow);
+
+            processedRows++;
+
+        }*/
+
+        // Append processed peak list to the project
+        project.addPeakList(processedPeakList);

```

Comment:
Need to check for isCanceled - as streams are not able to

` if (isCanceled())  return;`